### DISCUSSION TITLE
Problem with replacing "assign" with "set" when changing from liquid to nunjucks.

### DISCUSSION THREAD

**Discussion by arpeasal** [11ty/eleventy#2334] [Q&A] [answered]
NOTE: You can download this project as a zip file (about 21 MB, including node_modules) at (be sure to use http, NOT https-I'm not really using this server yet, so I haven't set up SSL certs.)
`http://rpallen.com.br/baixar/projeto-randall.zip`

I converted my project from liquid to nunjucks and I'm having a problem with changing liquid's "assign" to nunjucks's"set".  The project creates two sets of pages, one set of pages in which each page lists all posts that are in a single category, and the same thing for each tag.  The `/source/categoria/index.md` does this for the categories and `/source/tag/index.md` does it for the tags.

In `/source/categoria/index.md`, I changed "assign" to "set":
from: `{% assign artigos = collections.artigos | filterByCategory: categoria %}`
to: `{% set artigos = collections.artigos | filterByCategory: categoria %}`

This produced the error:
```
[11ty] > (./source/categoria/index.md) [Line 5, Column 58]
  expected block end in set statement
```

I did basically the same thing in `/source/tag/index.md`:
from: `{% set artigos = collections.artigos | filterByATag: tag %}`
to: `{% set artigos = collections.artigos | filterByATag: tag %} `

This produces a similar error:
```
[11ty] > (./source/tag/index.md) [Line 4, Column 54]
  expected block end in set statement
```

If I disable `/source/categoria/index.md` and `/source/tag/index.md` (by removing the ".md" extension from the index files), the site seems to work fine, except for, of course, clicking on a category or tag link doesn't take me to the appropriate page.  As expected, I get something like "Cannot GET /categorias/vocabulario/".

I can't figure out why it is expecting a "block end" or how to fix this error.

NOTE: Also, in both cases, I changed the line with the "reverse" filter from
`{% for artigo in artigos reversed %}`
to `{% for artigo in artigos | reverse %}`

I don't know for sure if this worked because the previous error kept me from seeing if it did work.

**Comment by pdehaan** [comment, marked as answer]
I didn't try downloading the 21 MB ZIP file, but remember that Nunjucks and LiquidJS use different syntaxes for passing values to filters.
See https://mozilla.github.io/nunjucks/templating.html#groupby

So Nunjucks uses parenthesis instead of colons, like:
```diff
-{% set artigos = collections.artigos | filterByATag: tag %}
+{% set artigos = collections.artigos | filterByATag(tag) %}
```

**Reply by arpeasal** [reply]
I discovered that syntax difference when I was looking for an answer yesterday and it didn't change anything.  When I tried that different syntax, I still got the same error.  I should have included that in my original post, but forgot that I'd tried that..

**Reply by arpeasal** [reply]
BTW, here's the entire code for the tag/index.md file - with the change in syntax for the filter:

```
---
pagination:
  data: collections.toda_tags
  size: 1
  alias: tag
permalink: "/tags/{{ tag | slug }}/index.html"
layout: layouts/pagina.njk
---

<h2>Artigos com o tag "{{ tag }}"</h2>

{% set artigos = collections.artigos | filterByATag(tag) %}

{% for artigo in artigos reversed %}
  <ul>
    <li>
    <a href="{{artigo.url}}">{{ artigo.data.title }}</a> - {{ artigo.data.date | readableDate }}:
     {{ artigo.data.trecho }}
    </li>
  </ul>
{% endfor %}
```

**Reply by pdehaan** [reply]
I'm not sure Nunjucks supports `{% for artigo in artigos reversed %}` syntax, and you might need to use the Nunjucks [`reverse`](https://mozilla.github.io/nunjucks/templating.html#reverse) filter:

```diff
-{% for artigo in artigos reversed %}
+{% for artigo in artigos | reverse %}
```

This worked locally for me:

```njk
---
pagination:
  data: collections.toda_tags
  size: 1
  alias: tag
permalink: "/tags/{{ tag | slug }}/index.html"
# layout: layouts/pagina.njk
---

<h2>Artigos com o tag "{{ tag }}"</h2>

{% set artigos = collections.artigos | filterByATag(tag) %}

{% for artigo in artigos | reverse %}
  <ul>
    <li>
    <a href="{{artigo.url}}">{{ artigo.data.title }}</a> - {{ artigo.date | readableDate }}:
     {{ artigo.data.trecho }}
    </li>
  </ul>
{% endfor %}
```

In fact, here's my test repo: https://github.com/pdehaan/11ty-2334

**Reply by arpeasal** [reply]
I actually know about the reverse issue.  I mentioned that at the very end of my original post.  I don't know why I didn't make that change in the copy of the code I posted, but I'm pretty sure it's correct in the project file zipped for download.

I went to the github site you linked (I know very little about using git and github).  I copied the raw code from the index.njk file and pasted it to my copy of the code.  It built ok, but it didn't work.  When I click on a category I get an error (such as: `Cannot GET /categorias/pronuncia`), not a page with the posts for that category.

**Reply by arpeasal** [reply]
Also, I saw that you commented out the layout in the front matter.  I tried it with that commented out and I also tried it with the layout applied.  Both gave the same result.

**Reply by pdehaan** [reply]
Your ZIP worked for me locally (on macOS 12.2), but I had to update the following files to work w/ Nunjucks format (ie: using `|reverse` filter and `filterByATag(tag)` syntax):
- source/categoria/index.md
- source/tag/index.md

**Reply by arpeasal** [reply]
I am so embarrassed.  I don't know how I zipped the wrong project.  I thought I zipped the one with the changes in it.

Anyway, I dowloaded the zip file, made the changes you mentioned and it built fine and performed as expected.  YEAH!!  But again, I am embarrassed and apologize for wasting your time with my foolish mistakes.
